definitions/marts/looker_studio_marts/ecf/ls_ecf_assurance_bad_shape_1_out_of_sequence.sqlx

```diff
@@ -0,0 +1,163 @@
+        3),('2021-09-01',
+        DATE_ADD(CURRENT_DATE(),INTERVAL 1 day),
+        'retained-3',
+        4),('2021-09-01',
+        DATE_ADD(CURRENT_DATE(),INTERVAL 1 day),
+        'retained-4',
+        5),('2025-08-01',
+        DATE_ADD(CURRENT_DATE(),INTERVAL 1 day),
+        'completed',
+        6),('2025-08-01',
+        DATE_ADD(CURRENT_DATE(),INTERVAL 1 day),
+        'extended-1',
+        7),('2025-08-01',
+        DATE_ADD(CURRENT_DATE(),INTERVAL 1 day),
+        'extended-2',
+        8),('2025-08-01',
+        DATE_ADD(CURRENT_DATE(),INTERVAL 1 day),
+        'extended-3',
+        9),('2021-09-01',
+        '2025-07-31',
+        'completed',
+        9),('2021-09-01',
+        '2025-07-31',
+        'extended-1',
+        6),('2021-09-01',
+        '2025-07-31',
+        'extended-2',
+        7),('2021-09-01',
+        '2025-07-31',
+        'extended-3',
+        8)]) AS declaration_sequence),
+  /* This identifies declarations received for a participant and induction details from their latest induction record. It joins in what should be the correct sequence for that declaration type based on when the declaration was received */
+  /* Only declarations eligible for funding are permitted for the calculation*/
+  extended_declarations AS (
+  SELECT
+    declaration_id,
+    declaration_type,
+    declaration_created_at_date,
+    declaration_date,
+    participant_profile_id,
+    state,
+    cpd_lead_provider_name,
+    statement_id,
+    schedule_identifier,
+    participant_type,
+    cohort,
+    statement_cohort,
+    trn,
+    declaration_sequence_list.expected_declaration_sequence_position,
+    /* This date logic in the 'order by' exists to ensure that if a declaration is out of order compared to declarations received earlier on a different date but also evidenced on the same day as a declaration that should be later in the sequence, that declaration won't eclipse the prior declaration and prevent the qualify statement from working at the end of this mart*/
+    /* For example, if you were to evidence a retained 3 declaration on Sep 15th and then on Sep 17th evidence a completed declaration at 9AM and then a retained 2 declaration at 10 AM,for the purposes of the queries below, you would want the retained 2 declaration ordered immediately after the retained 3 declaration and then the completed declaration ordered after the retained 2 declaration - So Ret 3, Ret 2, Completed */
+    /* This prevents this misordered retained 2 declaration from being excluded by the qualify statement excluding declarations out of order simply because they were evidenced on the same date, as this ordering logic compares it to the retained 3 declaration instead of the completed declaration. */
+    row_number() over (partition by participant_profile_id order by date(declaration_date) asc, expected_declaration_sequence_position asc) as actual_evidenced_sequence_position,
+  FROM
+    ${ref('ecf_declarations')} declarations
+  INNER JOIN

```

Comment:
Are you sure INNER and not LEFT? If you haven't specified the expected_declaration_sequence_position for a particular declaration_type do you want to find out when there's a NULL value in expected_declaration_sequence_position or a missing declaration that you don't necessarily know is there? Could potentially add an assertion to check for null values to handle this as well